.github/actions/version-matrix/src/pypi.ts

```diff
@@ -0,0 +1,73 @@
+import { compare } from 'semver';
+
+const pypiPackageInfoRoute = (pkg: string) => `https://pypi.org/pypi/${pkg}/json`;
+const pypiPackageVersionInfoRoute = (pkg: string, version: string) => `https://pypi.org/pypi/${pkg}/${version}/json`;
+
+// Only documents what we need
+interface PackageInfo {
+	info: {
+		requires_dist: string[];
+		requires_python: string;
+		version: string;
+		yanked: boolean;
+	};
+	releases: Record<
+		string,
+		{
+			python_version: string;
+			requires_python: string;
+			yanked: boolean;
+		}[]
+	>;
+}
+
+export async function fetchPackageVersions(packageName: string) {
+	const url = pypiPackageInfoRoute(packageName);
+
+	const response = await fetch(url);
+
+	if (!response.ok) {
+		throw new Error(`Failed to fetch package info for ${packageName}`, {
+			cause: await response.text(),
+		});
+	}
+
+	const json: PackageInfo = await response.json();
+
+	const rawVersions = Object.keys(json.releases);
+
+	// For some reason beta versions are `0.0.0a0` (where `a` is a "tag")
```

Comment:
Those are alphas, betas have b in them :slightly_smiling_face:

Reply:
🤷‍♂️, probably could've been better to say `tagged versions`!

```suggestion
	// For some reason tagged versions follow a structure like `0.0.0a0` (where `a` is a "tag")
```